discovery-provider/src/queries/get_user_playlist_update.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from typing import List
+
+from sqlalchemy.sql import text
+from src.utils import db_session
+
+logger = logging.getLogger(__name__)
+
+
+sql = text(
+    """
+SELECT
+    p.playlist_id,
+    p.updated_at,
+    ps.seen_at,
+    s.created_at
+from
+    playlists p
+INNER JOIN
+    saves s ON
+        s.save_item_id = p.playlist_id AND
+        s.is_current AND
+        NOT s.is_delete AND
+        s.save_type = 'playlist' AND
+        s.user_id = :user_id
+LEFT JOIN
+  playlist_seen ps ON 
+    ps.is_current AND
+    ps.playlist_id = p.playlist_id AND
+    ps.user_id = :user_id
+where
+    p.is_current = true AND
+    p.is_delete = false AND
+    s.created_at > p.updated_at
+"""
+)
+
+
+def get_user_playlist_update(user_id: int) -> List[int]:
+    """
+    Returns list of playlist_ids for a user_id which have been updated
+    and the user has not seen
+
```

Comment:
this returns playlists that have been updated both seen / unseen right? is it just that seen_at would be null for unseen updates?

Reply:
This returns playlists that have been updated. It may return null for the `seen_at` return because it was a left join with the playlist_seen table meaning that it is nullable. Note the application logic to handle if it is null in line 61 & 62

Playlist is created, then a user saves the playlist. 
From that point on, once the playlist is updated, this endpoint will return that playlist id as updated until the user "sees" the playlist. From which point this endpoint will not return it again until it is updated.

Reply:
yea i guess i was confused why that logic wasn't in the sql but makes sense either way